pkg/storage/tpr/install_types.go

```diff
@@ -45,20 +61,48 @@ func InstallTypes(cs clientset.Interface) error {
 		}
 
 		glog.Infof("Creating Third Party Resource Type: %s", tpr.Name)
-		if _, err := tprs.Create(&tpr); err != nil {
-			glog.Errorf("Failed to create Third Party Resource Type: %s (%s))", tpr.Name, err)
+		errChan = make(chan error, 1)
+		defer close(errChan)
+		errChans[tpr.Name] = errChan
+
+		execFunc(func(tpr v1beta1.ThirdPartyResource) error {
+			_, err := tprs.Create(&tpr)
 			return err
-		}
-		glog.Infof("Created TPR '%s'", tpr.Name)
-		// There can be a delay, so poll until it's ready to go...
-		for i := 0; i < 30; i++ {
-			if _, err := tprs.Get(tpr.Name); err != nil {
-				glog.Infof("TPR %s is ready", tpr.Name)
-				break
+		},
+			tpr,
+			errChans[tpr.Name],
+		)
+	}
+
+	errStr := ""
+	for name, ec := range errChans {
+		if err := <-ec; err != nil {
+			errStr = errStr + fmt.Sprintf("type: '%s', (%s)\n", name, err)
+		} else {
+			glog.Infof("Created TPR '%s'", name)
+
+			// There can be a delay, so poll until it's ready to go...
+			for i := 0; i < 30; i++ {
```

Comment:
This should use some form of the the upstream `wait.Poll`. 

As an example, you can check #459 where I use `wait.PollImmediate`.